### PR TITLE
RAB-1062: Handle SFTP login with private key

### PR DESCRIPTION
**Pull request by axelhms**
**Description (for Contributor and Core Developer)**

Add support for SFTP login with private key

PR back: https://github.com/akeneo/pim-enterprise-dev/pull/15248

### CODE REVIEW

File: src/Akeneo/Platform/Bundle/ImportExportBundle/front/src/components/StorageForm.test.tsx [hunk 1]
```diff
@@ -1,10 +1,15 @@
...
+import {screen, act} from '@testing-library/react';
 import {renderWithProviders, ValidationError, useFeatureFlags} from '@akeneo-pim-community/shared';
 import {StorageForm} from './StorageForm';
 import {LocalStorage, NoneStorage, SftpStorage} from './model';
 
+global.fetch = jest.fn().mockImplementation(async () => ({
+  ok: true,
+  json: async () => '-----BEGIN CERTIFICATE-----publickey-----END CERTIFICATE-----',
+}));
+
```
Comment (StevenVAIDIE) [inline src/Akeneo/Platform/Bundle/ImportExportBundle/front/src/components/StorageForm.test.tsx:12]: We didn't need it now as we do it in beforeEach

**Review by StevenVAIDIE — APPROVED** [review]

File: src/Akeneo/Platform/Bundle/ImportExportBundle/front/src/components/StorageConfigurator/SftpStorageConfigurator.tsx [hunk 2]
```diff
@@ -31,6 +83,10 @@ const SftpStorageConfigurator = ({
   const translate = useTranslate();
   const portValidationErrors = filterErrors(validationErrors, '[port]');
   const [isValid, canCheckConnection, checkReliability] = useCheckStorageConnection(storage);
+  const publicKey = useGetPublicKey();
+
+  const canCopyToClipboard = (): boolean | undefined => ('clipboard' in navigator ? true : undefined);
```
Comment (ValentinMumble) [inline src/Akeneo/Platform/Bundle/ImportExportBundle/front/src/components/StorageConfigurator/SftpStorageConfigurator.tsx:88]:
```suggestion
  const canCopyToClipboard = (): boolean => 'clipboard' in navigator;
```
?

File: src/Akeneo/Platform/Bundle/ImportExportBundle/front/src/components/StorageConfigurator/SftpStorageConfigurator.tsx [hunk 3]
```diff
@@ -83,6 +139,23 @@ const SftpStorageConfigurator = ({
...
+          clearable={false}
+        >
+          {STORAGE_LOGIN_TYPES.map(loginType => (
+            <SelectInput.Option value={loginType} key={loginType}>
+              {translate(`pim_import_export.form.job_instance.storage_form.connection_type.${loginType}`)}
+            </SelectInput.Option>
+          ))}
+        </SelectInput>
+      </Field>
+
```
Comment (ValentinMumble) [inline src/Akeneo/Platform/Bundle/ImportExportBundle/front/src/components/StorageConfigurator/SftpStorageConfigurator.tsx:158]:
```suggestion
```

File: src/Akeneo/Platform/Bundle/ImportExportBundle/front/src/components/StorageConfigurator/SftpStorageConfigurator.tsx [hunk 4]
```diff
@@ -91,15 +164,25 @@ const SftpStorageConfigurator = ({
...
+          required={true}
+          type="password"
+          label={translate('pim_import_export.form.job_instance.storage_form.password.label')}
+          onChange={(password: string) => onStorageChange({...storage, password})}
+          errors={filterErrors(validationErrors, '[password]')}
+        />
+      ) : (
+        <Field label={translate('pim_import_export.form.job_instance.storage_form.public_key.label')}>
+          <CopyableInputContainer>
+            <CopyableInput disabled value={publicKey ?? ''} />
```
Comment (ValentinMumble) [inline src/Akeneo/Platform/Bundle/ImportExportBundle/front/src/components/StorageConfigurator/SftpStorageConfigurator.tsx:180]:
```suggestion
            <CopyableInput disabled={true} value={publicKey ?? ''} />
```

File: src/Akeneo/Platform/Bundle/ImportExportBundle/front/src/components/StorageForm.test.tsx (hunk 1 above)
Comment (ValentinMumble) [inline src/Akeneo/Platform/Bundle/ImportExportBundle/front/src/components/StorageForm.test.tsx:12]:
```suggestion
```

**Review by ValentinMumble — APPROVED** [review]

**Pull request merged**